### ISSUE TITLE
Tutorial and Download Links Missing

### ISSUE THREAD

**Issue by hauadorth** [frno7/linux#87]
How do i install them i dont understand anything and the download links are missing

**Comment by frno7** [comment]
Yeah, Git Hub apparently removes builds after a few months, sadly. I updated the main branch. Let’s see if it rebuilds successfully. :-)

**Comment by DCFUKSURMOM** [comment]
GitHub moment

**Comment by frno7** [comment]
Git Hub produced some downloadable [artifacts](https://github.com/frno7/linux/actions/runs/5715491404) just now!

**Comment by AKuHAK** [comment]
Is it possible to add NTSC binary into releases page?

**Comment by frno7** [comment]
@AKuHAK, I just added NTSC to

https://github.com/frno7/linux/blob/a9554161590f2b0fd39451e9924c9e7f3c12ed5d/.github/workflows/compilation.yml#L98-L102

Now let’s see if this thing builds.. :-)

**Comment by frno7** [comment]
With

- [initramfs-ntsc.tar.gz](https://github.com/frno7/linux/releases/download/ps2-main-package/initramfs-ntsc.tar.gz)
- [initramfs-pal.tar.gz](https://github.com/frno7/linux/releases/download/ps2-main-package/initramfs-pal.tar.gz)
- [vmlinuz-ntsc-ps2-main-a9554161.elf](https://github.com/frno7/linux/releases/download/ps2-main-package/vmlinuz-ntsc-ps2-main-a9554161.elf)
- [vmlinuz-pal-ps2-main-a9554161.elf](https://github.com/frno7/linux/releases/download/ps2-main-package/vmlinuz-pal-ps2-main-a9554161.elf)

it seems we’ve now got a combined PAL and NTSC release. :-)

**Comment by Ghost-Bit** [comment]
any way you could upload an unmodified vmlinuz for anyone more comfortable with kloader? the page phrases it as if not using kloader is optional when in fact it is currently impossible to use kloader with the files provided since the vmlinuz is baked into an executable... or is it impossible to use kloader, and that's why it's contained within a special executable? in which case, i propose a change to denote that the kernel won't work with it

**Comment by frno7** [comment]
Hi @Ghost-Bit. Oh, [kernelloader](https://kernelloader.sourceforge.net/linux.html)? I’ve only used it with Linux 2.x, and that was some 10 years ago or so. I don’t know if kernelloader can be made to work with Linux 5.x.

Linux 3.x, 4.x and 5.x are made for [wLaunchELF](https://github.com/ps2homebrew/wLaunchELF), which I believe is much faster and reliable. These later Linux versions have significant improvements over 2.x. For example, loading IOP modules on-demand by themselves, don’t rely on any Sony BIOS blobs, much less use of Sony IOP blobs, and so on.

**Comment by DCFUKSURMOM** [comment]
being able to use kernelloader would be cool, would let people tweak command line options without recompiling. will this kernel build with a normal mips toolchain? it may work in kernelloader if its built as a normal kernel image and not as a ps2 executable

**Comment by DCFUKSURMOM** [comment]
i wouldnt be able to test it myself, my ps2 isnt in one piece and kernelloader does not work in pcsx2 last time i checked

**Comment by Ghost-Bit** [comment]
_would let people tweak command line options without recompiling. will this kernel build with a normal mips toolchain? it may work in kernelloader if its built as a normal kernel image and not as a ps2 executable_
this is the main reason i ask as an owner of a Phat, i don't want to use a flash drive, i want to leverage the full ability of the hardware i have

**Comment by frno7** [comment]
> it may work in kernelloader if its built as a normal kernel image and not as a ps2 executable

Linux 3.x, 4.x, and 5.x are compiled with [`make vmlinuz`](https://github.com/frno7/linux/wiki/Building-a-PlayStation-2-Linux-kernel), which is about as “normal” as it can possible get, no? Normal Linux kernels are normally [ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format). :-)

**Comment by DCFUKSURMOM** [comment]
I know they are normally elf, but doesn't this have some some extra stuff? I could be misremembering, it's been a while since I've had a usable PS2 lol.

**Comment by frno7** [comment]
Hmm... commit d196e9d3fa9fab341499ef9c2ca83607a28d646c is the only boot fix I recall. It should be harmless, though. Maybe kernelloader needs something extra that normal `vmlinuz` doesn’t have?

I don’t recall the MIPS architecture (as a whole, not only the PlayStation 2) having a standard method to pass command line options. Maybe that and some other extensions were made for the 2.x kernel in combination with kernelloader? I remember that kernelloader preloaded a bunch of IRX files too, but as mentioned, that’s no longer necessary (and won’t even work), because modern Linux reboots the IOP and loads IRX files on-demand, as needed, by itself. I guess that IRX feature of kernelloader is optional, and can be safely ignored though?

**Comment by DCFUKSURMOM** [comment]
I don't think the old 2.x kernels could be loaded as a PS2 executable directly? Pretty sure the external loader was a hard requirement.

**Comment by Ghost-Bit** [comment]
If nothing else can be done with the possibility of making this work with kloader, a method of inbuilt external configuration loading would be nice as I believe most unfamiliar with or slow to learn build systems will generally frown upon any program with hardcoded configuration especially given the chance of coming across incompatibilities in their toolchain or even difficulty setting up a toolchain given the OS they use (I cannot imagine most Windows users will understand or complete anything in this guide without several days of work setting up msys or temporarily switching to Linux directly and suffering with not understanding anything about Linux either)

even if it's loaded from a hardcoded location like the first memory card I do feel like this is an important quality of life thing

**Comment by frno7** [comment]
> I don't think the old 2.x kernels could be loaded as a PS2 executable directly? Pretty sure the external loader was a hard requirement.

Linux 2.x are unable to manage the [IOP](https://github.com/frno7/linux/wiki/Frequently-asked-questions-about-PlayStation-2-Linux#what-is-the-iop), so kernelloader is needed for things like that. Linux 2.x also rely on the Sony BIOS for vital services. I believe these are the main reasons Linux 2.x won’t load “directly” on the metal, as opposed to Linux 5.x that handle (almost) all hardware by themselves, including the IOP (and its IRX files), and discard the Sony BIOS to reclaim that memory.

@Ghost-Bit, yeah, there are a lot of features one might want to have, in addition to a more flexible configuration. For example audio (issue #16) and video (issue #10), floating-point (issue #3), networking (issue #19), harddisk (issue #18), etc. :-)

**Comment by DCFUKSURMOM** [comment]
so i was thinking about this and i think i remembered what kernelloader does not like, i think it does not like the combined kernel and initrd, i think if you build the kernel and initrd separately instead of building the initrd into the kernel it may work
